### ISSUE TITLE
Tenant ls command does not works if vm that was added to tenant is deleted.

### ISSUE THREAD

**Issue by ashahi1** [vmware/docker-volume-vsphere#929]
Steps to reproduce:

1) Create a tenant and associate a vm to it:
  /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py tenant create --name=tenant1 --vm-list=photon4
2) Tenant ls command gives following output:
```
[root@sc-rdops-vm06-dhcp-186-90:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py tenant ls
Uuid                                            Name                   Description               Default_datastore  VM_list
------------------------------------  --------  ------------------------  -----------------  -------
41242167-6019-4f3f-a384-66a82cf9997d   _DEFAULT   This is a default tenant
5aa44411-0eba-4591-bacf-c4cfc27beb85    tenant1                                                      Photon4
```
3) Delete the vm that was added to the tenant.
4) Tenant ls command gives following output - 

[root@sc-rdops-vm06-dhcp-186-90:~] /usr/lib/vmware/vmdkops/bin/vmdkops_admin.py tenant ls
Traceback (most recent call last):
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 1004, in <module>
    main()
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 51, in main
    args.func(args)
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 862, in tenant_ls
    rows = generate_tenant_ls_rows(tenant_list)
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 810, in generate_tenant_ls_rows
    vm_list = generate_vm_list(tenant.vms)
  File "/usr/lib/vmware/vmdkops/bin/vmdkops_admin.py", line 790, in generate_vm_list
    res = res + vm_name
TypeError: Can't convert 'NoneType' object to str implicitly

============
Ideally the tenant ls command should give the list of all the tenants with deleted vm being no longer part of the VM_list for that particular tenant.

**Comment by govint** [comment]
PR #937 addresses a small fix to prevent this issue.

The larger issue is the tenant DB needs to be updated to remove VMs (uuids) that have since been deleted. Possibly separate issue for that.

**Comment by govint** [comment]
OTOH, if a VM is added to a tenant from host ESX1 and then from another host ESX2 we do "ls" with admin CLI then too we will not find the VM name given its UUID. Becasue the VM is not known to host ESX2. But the VM is very much there on the host ESX1.

In which case the UUID should get displayed.

These multi-host scenarios don't play well with VM names and there isn;t a way to "clean" tenants of VMs that are really deleted. We can never know on an ESX host that a VM is deleted, except on the local host.

**Comment by pdhamdhere** [comment]
@govint @shaominchen 

>The larger issue is the tenant DB needs to be updated to remove VMs (uuids) that have since been deleted. Possibly separate issue for that.

Does [PropertyCollector](http://pubs.vmware.com/vsphere-65/index.jsp#com.vmware.wssdk.pg.doc/PG_PropertyCollector.7.1.html) help to detect
- VM re-name change and update KV
- VM is deleted and update tenant DB
- VM is migrated and update tenant DB (need to confirm)

**Comment by govint** [comment]
Fixed via #937
